integration-libs/s4-service/order/components/s4-service-order-detail-actions/s4-service-order-detail-actions.component.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, inject, OnDestroy, OnInit } from '@angular/core';
+import { GlobalMessageService, GlobalMessageType } from '@spartacus/core';
+import { OrderDetailActionsComponent } from '@spartacus/order/components';
+import { Order } from '@spartacus/order/root';
+import { CheckoutServiceSchedulePickerService } from '@spartacus/s4-service/root';
+import { Subject, takeUntil } from 'rxjs';
+
+@Component({
+  selector: 'cx-s4-service-order-detail-actions',
+  templateUrl: './s4-service-order-detail-actions.component.html',
+})
```

Comment:
```suggestion
@Component({
  selector: 'cx-s4-service-order-detail-actions',
  templateUrl: './s4-service-order-detail-actions.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
```
`major:` let's keep the convention and add change detection strategy OnPush to prevent possible redundant re-rendering of this component and its child.

Reply:
Keeping the change detection strategy as `OnPush` in this component wrongly shows or hides the action button.

Reply:
It might be related to the local state of this component (`order` and `displayActions` that are not reactive). Let's try using `order$` directly in the component and change `displayActions$` to reactive stream:
```ts
@Component({
  selector: 'cx-s4-service-order-detail-actions',
  templateUrl: './s4-service-order-detail-actions.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class S4ServiceOrderDetailActionsComponent extends OrderDetailActionsComponent {
  private checkoutServiceSchedulePickerService = inject(
    CheckoutServiceSchedulePickerService
  );
  private globalMessageService = inject(GlobalMessageService);

  displayActions$: Observable<boolean> = this.order$.pipe(
    map((order) => this.checkServiceStatus(order))
  );

  checkServiceStatus(order: Order): boolean {
    if (order && order.status === 'CANCELLED') {
      return false;
    } else if (order && order.servicedAt) {
      const hoursFromSchedule =
        this.checkoutServiceSchedulePickerService.getHoursFromServiceSchedule(
          order.servicedAt
        );
      if (hoursFromSchedule > 0 && hoursFromSchedule <= 24) {
        this.globalMessageService.add(
          { key: 'rescheduleService.serviceNotAmendable' },
          GlobalMessageType.MSG_TYPE_INFO
        );
        return false;
      } else if (hoursFromSchedule > 24) {
        return true;
      }
    }
    return true;
  }
}
```

Then we need to adjust the template:
```html
<ng-container *ngIf="order$ | async as order">
  <div class="cx-order-details-actions row" *ngIf="displayActions$ | async">
   <!--  -->
  </div>
</ng-container>
```
Let's try this approach and let me know whether business requirements are met.